doc/tutorial/03_concepts/model_a_concept.cpp

```diff
@@ -0,0 +1,29 @@
+#include <cmath>
+
+#include <seqan3/core/debug_stream.hpp>
+
+//![concept]
+// helper concept has_foo:
+template <typename T>
+concept has_foo = requires (T val) {
+                      typename T::FOO; // requirement 1
+                      val.foo;         // requirement 2
+                  };
+
+// concept fooger:
+template <typename T>
+concept fooger = has_foo<T> && std::same_as<typename T::FOO, int>;
+//![concept]
+
+struct my_type
+{
+    using FOO = int;
+    char foo; // foo can be of any type, here it is of type `char`
+};
+
+//![main]
+int main()
+{
+    seqan3::debug_stream << fooger<my_type> << std::endl; // should print 1
```

Comment:
```suggestion
    seqan3::debug_stream << std::boolalpha << fooger<my_type> << '\n'; // should print true
```

`std::boolalpha` should work with our debug_stream?
Not sure if we want it :)

Reply:
Although the output is nicer I think it makes the snippet more complex than it needs to be